cmd/soroban-cli/src/config/sign_with.rs

```diff
@@ -0,0 +1,120 @@
+use std::path::PathBuf;
+
+use crate::{
+    signer::{self, Stellar},
+    xdr::{Transaction, TransactionEnvelope},
+};
+use clap::arg;
+
+use super::{
+    locator,
+    network::{self, Network},
+    secret::{self, StellarSigner},
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Network(#[from] network::Error),
+    #[error(transparent)]
+    Signer(#[from] signer::Error),
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    Locator(#[from] locator::Error),
+    #[error(transparent)]
+    Rpc(#[from] soroban_rpc::Error),
+    #[error("No sign with key provided")]
+    NoSignWithKey,
+}
+
+#[derive(Debug, clap::Args, Clone, Default)]
+#[group(skip)]
+pub struct Args {
+    /// Sign with account. Alias `source`. Can be an identity (--source alice), a secret key (--source SC36…), or a seed phrase (--source "kite urban…").
+    #[arg(
+        long,
+        conflicts_with = "sign_with_laboratory",
+        env = "STELLAR_SIGN_WITH_SECRET"
+    )]
+    pub sign_with_key: Option<String>,
+    /// Sign with labratory
+    #[arg(
+        long,
+        visible_alias = "sign-with-lab",
+        conflicts_with = "sign_with_key",
+        env = "STELLAR_SIGN_WITH_LABRATORY"
+    )]
+    pub sign_with_laboratory: bool,
+
+    #[arg(long, conflicts_with = "sign_with_laboratory")]
+    /// If using a seed phrase, which hierarchical deterministic path to use, e.g. `m/44'/148'/{hd_path}`. Example: `--hd-path 1`. Default: `0`
+    pub hd_path: Option<usize>,
+
+    /// If `--sign-with-*` is used this will remove requirement of being prompted to sign transactions
+    #[arg(long)]
+    pub yes: bool,
+
+    #[command(flatten)]
+    pub network: network::Args,
+
+    #[command(flatten)]
+    pub locator: locator::Args,
+}
+
+impl Args {
+    pub fn signer(&self) -> Result<StellarSigner, Error> {
+        let account = self.sign_with_key.as_deref().ok_or(Error::NoSignWithKey)?;
+        Ok(self
+            .locator
+            .account(account)?
+            .signer(self.hd_path, !self.yes)?)
+    }
+
+    pub async fn sign_txn_env(
```

Comment:
IMO this 2 methods should go into StellarSigner

Reply:
These methods are so that you don't need to provide the signer. Hence `self.signer`, and then the next one uses `self.get_network`. So in the first case we are looking up which signer to use. And the second which network. This is so the StellarSigner has the context.

Reply:
But we can move `sign` to signer and add one more argument which is a network?  Don't have a strong preference here, it's just for me it's not clear right now what's the logical separation between `config::Args` and `Signer` is. It feels like `Signer` should do the job, while `config::Args` just provide entities that do the job. But we can always re-visit this later

Reply:
The idea here is that the `self.signer` for `config.arg` could be from `source_account`, where as for `sign_with` it is the one provided by `--sign-with-*`. So this is just a connivence method that will use the `signer` resolved here instead of the one used in `sign_with::Args`. This is all so that the commands don't need to resolve both the signer and the network to sign. But we could have it that way. Again I think this should be address in a future PR as it wouldn't change the CLI's UX.

Reply:
I'd say it's more of a nitpick, so fine to leave as is for now, considering size of PR